kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/encrypt/InBandEncryptionManager.java

```diff
@@ -102,13 +104,18 @@ private ByteBufferOutputStream allocateBufferForEncrypt(@NonNull MemoryRecords r
         return bufferAllocator.apply(sizeEstimate);
     }
 
-    private CompletionStage<MemoryRecords> attemptEncrypt(@NonNull String topicName,
+    private CompletionStage<MemoryRecords> attemptEncrypt(@NonNull String virtualClusterName,
+                                                          @NonNull String topicName,
                                                           int partition,
                                                           @NonNull EncryptionScheme<K> encryptionScheme,
                                                           @NonNull MemoryRecords records,
                                                           int attempt,
                                                           @NonNull IntFunction<ByteBufferOutputStream> bufferAllocator,
                                                           int allRecordsCount) {
+        var encyptedRecordsTotal = RecordEncryptionMetrics.KROXYLICIOUS_ENCRYPTED_RECORDS_BASE_METER_NAME_TOTAL_METER_PROVIDER
```

Comment:
I planned to count the encrypted record here since it provides me the total record that are encrypted successfully but I can move this to `RecordEncryptionFilter` class itself. I was worried to initialize it in `RecordEncryptionFilter` class because I see that in the `maybeEncodeProduce()` method, it was possible that the Encryption failed somehow and I incremented the metrics even when the encryption failed